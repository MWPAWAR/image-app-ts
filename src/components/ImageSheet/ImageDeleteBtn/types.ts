import { ImageTypes } from '../../types'

export interface ImageDeleteBtnProps {
  selectedImage: ImageTypes
}
