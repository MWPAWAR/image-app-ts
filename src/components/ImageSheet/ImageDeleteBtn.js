import { useDispatch } from 'react-redux';
import imageDeleteBtnStyles from './ImageDeleteBtn.module.css';
import { deleteImg } from '../../store/images/images';
import { resetSelectedImg } from '../../store/imageUiState/imageUiState';

const ImageDeleteBtn = ({ selectedImage }) => {
  const dispatch = useDispatch();
  const handleDeleteClick = () => {
    dispatch(deleteImg(selectedImage.id));
    dispatch(resetSelectedImg(selectedImage.id));
  };

  return (
    <div className={imageDeleteBtnStyles.deleteBtnFooter}>
      <button
        className={imageDeleteBtnStyles.button}
        onClick={handleDeleteClick}
        >Delete</button>
    </div>
  )
};

export default ImageDeleteBtn;
