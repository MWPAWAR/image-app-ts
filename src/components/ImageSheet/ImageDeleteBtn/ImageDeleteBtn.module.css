.button {
  background-color: #fff;
  width: 100%;
  padding: 10px;
  border: 1px solid #dadce0;
  cursor: pointer;
}

.deleteBtnFooter {
  margin-top: 30px;
}
