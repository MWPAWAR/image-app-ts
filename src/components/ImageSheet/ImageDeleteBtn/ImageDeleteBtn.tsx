import { useAppDispatch } from '../../customHooks/hooks'
import imageDeleteBtnStyles from './ImageDeleteBtn.module.css'
import { deleteImg } from '../../../store/images/images'
import { resetSelectedImg } from '../../../store/imageUiState/imageUiState'
import { ImageDeleteBtnProps } from './types'
import { getString } from '../../../i18n'

const ImageDeleteBtn = ({ selectedImage }: ImageDeleteBtnProps) => {
  const dispatch = useAppDispatch()
  const handleDeleteClick = (): void => {
    dispatch(deleteImg({ id: selectedImage.id }))
    dispatch(resetSelectedImg())
  }

  return (
    <div className={imageDeleteBtnStyles.deleteBtnFooter}>
      <button
        className={imageDeleteBtnStyles.button}
        onClick={handleDeleteClick}
      >
        {getString('imageSheet.deleteBtnLabel')}
      </button>
    </div>
  )
}

export default ImageDeleteBtn
