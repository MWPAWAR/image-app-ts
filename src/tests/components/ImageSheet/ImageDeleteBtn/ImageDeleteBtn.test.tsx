import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ImageDeleteBtn from '../../../../components/ImageSheet/ImageDeleteBtn/ImageDeleteBtn'
import { imageMock1 } from '../../../mocks/image'

describe('ImageDeleteBtn', () => {
  const mockStore = configureStore([])
  const store = mockStore({})
  jest.spyOn(store, 'dispatch')

  const ImageDeleteBtnComponent = () => (
    <MemoryRouter>
      <Provider store={store}>
        <ImageDeleteBtn selectedImage={imageMock1} />
      </Provider>
    </MemoryRouter>
  )

  it('Matches ImageDeleteBtn snapshot', () => {
    const tree = renderer.create(<ImageDeleteBtnComponent />)
    expect(tree).toMatchSnapshot()
  })

  it('Calls dispatch twice if user clicks on delete button', async () => {
    const { getByText } = render(<ImageDeleteBtnComponent />)
    await userEvent.click(getByText('Delete'))
    expect(store.dispatch).toBeCalledTimes(2)
    expect(store.dispatch).toBeCalledWith({
      type: 'imageUiState/resetSelectedImg',
    })
    expect(store.dispatch).toBeCalledWith({
      type: 'images/deleteImg',
      payload: { id: imageMock1.id },
    })
  })
})
