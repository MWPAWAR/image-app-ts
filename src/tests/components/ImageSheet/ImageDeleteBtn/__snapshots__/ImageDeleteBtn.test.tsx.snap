// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDeleteBtn Matches ImageDeleteBtn snapshot 1`] = `
<div
  className="deleteBtnFooter"
>
  <button
    className="button"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
